code/game/turfs/change_turf.dm

```diff
@@ -108,24 +108,26 @@ GLOBAL_LIST_INIT(blacklisted_automated_baseturfs, typecacheof(list(
 	return W
 
 // Take off the top layer turf and replace it with the next baseturf down
-/turf/proc/ScrapeAway()
+/turf/proc/ScrapeAway(amount=1, flags)
+	if(!amount)
+		return
 	if(length(baseturfs))
-		var/list/new_baseturfs = baseturfs
-		var/turf_type = new_baseturfs[new_baseturfs.len]
-		new_baseturfs.len--
-		switch(new_baseturfs.len)
-			if(1)
-				new_baseturfs = new_baseturfs[1]
-			if(0)
-				new_baseturfs = turf_type
-				// We must never end up with a situation where there is no baseturf
-				WARNING("turf of type [type] had a baseturfs length 1 still in list form.")
-		return ChangeTurf(turf_type, new_baseturfs)
+		var/list/new_baseturfs = baseturfs.Copy()
+		var/turf_type = new_baseturfs[max(1, new_baseturfs.len - amount + 1)]
+		while(ispath(turf_type, /turf/baseturf_skipover))
```

Comment:
Wouldn't a string define be better here ? Faster and you can drop the fake path ?

Reply:
I did this primarily for 2 reasons, every other value in the list is a turf path and if anything grabs a baseturf value to instantiate it would runtime on text.

Additionally, we may actually *want* to instantiate skipover turfs to run some code in initialize when some border is reached. Also this way we can put more data into the path to be accessed by initial() if need be.

I felt the costs were worth the possible future benefits but in the end it doesn't really matter too much.